week-4/E-arrays-of-objects/exercise-3.js

```diff
@@ -54,32 +54,57 @@ DO NOT EDIT ANYTHING ABOVE THIS LINE
 WRITE YOUR CODE BELOW
 */
 
-
 var restaurantFinderApplication = {
-    applicationName: "Restaurant Finder",
-    applicationVersion: "1.0",
-    restaurants: restaurants,
-    findAvailableRestaurants: function(numberOfPeople) {
-        // Complete here
-    },
-    findRestaurantServingDish: function(dishName) {
-        // Complete here
-    },
-    countNumberOfRestaurantsInArea: function(area) {
-        // Complete here
-    }
+  applicationName: "Restaurant Finder",
+  applicationVersion: "1.0",
+  restaurants: restaurants,
+  findAvailableRestaurants: function(numberOfPeople) {
+    return restaurants
+      .filter(
+        restaurant =>
+          restaurant.totalSeats - restaurant.numberOfCustomers > numberOfPeople
+      )
+      .map(displayName);
+  },
+  findRestaurantServingDish: function(dishName) {
+    return restaurants
+      .filter(restaurant => restaurant["menu"].includes(dishName))
+      .map(displayName);
+  },
+  countNumberOfRestaurantsInArea: function(area) {
+    return restaurants.filter(restaurant => restaurant.address.area === area)
+      .length;
+  }
 };
 
+function displayName(restaurant) {
+  return restaurant.name;
+}
 
```

Comment:
Well done - great use of a helper function `displayName()` to avoid duplicating the logic. Good DRY principles!!

Reply:
Thank you.